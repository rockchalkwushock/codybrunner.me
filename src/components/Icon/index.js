import React from 'react'
import styled from 'styled-components'
import { string } from 'prop-types'

const StyledIcon = styled.i`
  color: ${props => props.theme.site.bg};
`

StyledIcon.displayName = 'Icon'

const SiteIcon = ({ className }) => <StyledIcon className={className} />

SiteIcon.propTypes = {
  className: string.isRequired
}

export default SiteIcon
