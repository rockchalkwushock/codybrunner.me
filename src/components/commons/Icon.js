import React from 'react'
import styled from 'styled-components'
import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faGithub,
  faInstagram,
  faLinkedin,
  faMediumM,
  faTwitter,
  faYoutube
} from '@fortawesome/free-brands-svg-icons'
import { faBars, faSpinner } from '@fortawesome/free-solid-svg-icons'

// By setting things up like this react-fontawesome knows of the icons
// I am also only using these icons from the libraries so less overhead.
library.add(
  faGithub,
  faInstagram,
  faLinkedin,
  faMediumM,
  faTwitter,
  faYoutube,
  faBars
)

const UnstyledIcon = props => <FontAwesomeIcon {...props} />

export const StyledIcon = styled(UnstyledIcon)`
  color: ${({ nav, theme }) =>
    nav === 'true' ? theme.colors.alt : theme.colors.secondary};
  font-size: 1.75rem;
  margin: 0 0.5rem;

  :hover {
    color: yellow;
  }

  @media (min-width: ${({ theme }) => theme.screen.large}) {
    font-size: 2em;
  }
`

const SiteIcon = ({ nav, icon, size }) => (
  <StyledIcon nav={nav} icon={icon} size={size} />
)

const StyledLoader = styled.div`
  align-items: center;
  color: ${({ theme }) => theme.colors.secondary};
  display: flex;
  flex-direction: column;
  justify-content: center;
`

const Loading = () => (
  <StyledLoader>
    <FontAwesomeIcon icon={faSpinner} size="3x" spin />
  </StyledLoader>
)

export { Loading }
export default SiteIcon
