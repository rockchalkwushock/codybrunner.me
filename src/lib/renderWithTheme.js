import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
