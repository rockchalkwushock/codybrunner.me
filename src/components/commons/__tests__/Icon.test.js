import React from 'react'
import { shallow } from 'enzyme'

import renderWithTheme from '../../../lib/renderWithTheme'
import Icon, { StyledIcon } from '../Icon'

describe('Component: <Icon />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Icon className="fab fa-twitter" />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have props', () => {
    expect(wrapper.props()).toHaveProperty('className', 'fab fa-twitter')
    expect(wrapper.props()).toHaveProperty('size', 1.75)
  })
  test('should have new size', () => {
    expect(wrapper.props()).toHaveProperty('size', 1.75)
    wrapper.setProps({ size: 2 })
    wrapper.update()
    expect(wrapper.props()).toHaveProperty('size', 2)
  })
})

describe('Component: <StyledIcon />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<StyledIcon className="fab fa-twitter" />)
  })
  test('should render correclty', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('color', '#143a52')
    expect(wrapper).toHaveStyleRule('font-size', '1.75em')
    expect(wrapper).toHaveStyleRule('transition', 'all 0.2s ease-in-out')
    expect(wrapper).toHaveStyleRule('color', '#ffe1b6', {
      modifier: ':hover'
    })
    expect(wrapper).toHaveStyleRule('font-size', '2em', {
      media: '(min-width:736px)'
    })
  })
})
