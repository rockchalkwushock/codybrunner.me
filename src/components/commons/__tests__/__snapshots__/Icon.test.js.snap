// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Icon /> should render correctly 1`] = `
<Icon
  className="fab fa-twitter"
  size={1.75}
/>
`;

exports[`Component: <StyledIcon /> should render correclty 1`] = `
.c0 {
  color: #143a52;
  font-size: 1.75em;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  color: #ffe1b6;
}

@media (min-width:736px) {
  .c0 {
    font-size: 2em;
  }
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "fg": "",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
      },
    }
  }
>
  <Icon
    className="fab fa-twitter"
    size={1.75}
  >
    <svg
      className="fab fa-twitter c0"
      size={1.75}
    />
  </Icon>
</ThemeProvider>
`;
